e2e/test/iothub/service/RegistryE2ETests.cs

```diff
@@ -119,7 +119,13 @@ public async Task DevicesClient_AddDeviceWithTwinWithDeviceCapabilities()
 
             try
             {
-                Device actual = await serviceClient.Devices.GetAsync(deviceId).ConfigureAwait(false);
+                Device actual = null;
+                do
+                {
+                    // allow some time for the device registry to update the cache
+                    await Task.Delay(50).ConfigureAwait(false);
+                    actual = await serviceClient.Devices.GetAsync(deviceId).ConfigureAwait(false);
+                } while (actual == null);
```

Comment:
Don't we have a "retry this operation" helper method for this kind of thing?

Reply:
Yeah, but it takes more lines than this and I think this is more readable.